lib/puppet/provider/gluster_volume/gluster_volume.rb

```diff
@@ -25,6 +25,13 @@ def self.prefetch(resources)
     end
   end
 
+  def missing_peers
+    peers = self.class.all_peers
+    [:hostname, :fqdn, :ipaddress].each { |f| peers << Facter.value(f) }
```

Comment:
Can you explain this line please? `peers` is an array of what? Then you just add Facter values? I'm a bit confused..